src/Microsoft.Health.Fhir.Core/Features/Definition/ISearchParameterDefinitionManager.cs

```diff
@@ -63,5 +63,13 @@ public interface ISearchParameterDefinitionManager
         /// <param name="componentIndex">The optional component index if the search parameter is a composite</param>
         /// <returns>The search parameter type.</returns>
         SearchParamType GetSearchParameterType(SearchParameterInfo searchParameter, int? componentIndex);
+
+        /// <summary>
+        /// Gets list of search parameters with the given support status: <paramref name="isSupported"/> <paramref name="isSearchable"/>.
+        /// </summary>
+        /// <param name="isSupported">Indicates whether to return supported params or unsupported params.</param>
+        /// <param name="isSearchable">Indicates whether to return searchable or unsearchable params.</param>
+        /// <returns>An <see cref="IEnumerable{T}"/> that contains the matching search parameters.</returns>
+        IEnumerable<SearchParameterInfo> GetSearchByStatus(bool isSupported, bool isSearchable);
```

Comment:
Perhaps this method is a bit awkward?  Looking at it again, I am thinking that perhaps we should split up the interface and include this only on the SupportedSearchParameterDefinitionManager?  Please leave your thoughts.  Thanks.

Reply:
Right, `SearchableSearchParameterDefinitionManager.AllSearchParameters` will return an `IEnumerable<SearchParameterInfo>` of all the search parameters that are marked as `isSearchable` and `SupportedSearchParameterDefinitionManager` will do the same for `isSupported`. It might be nice to piggy back off of the existing functionality or replace it. I just doubled checked, the filtering classes aren't used anywhere in the solution yet - but it might be good to check with @brendankowitz if you are thinking of removing them.

Reply:
I updated the interface and added a method only to the SupportedSearchParameterDefinitionManager.  Thanks for the review.